docs/api-tutorials/custom-reporter.md

```diff
@@ -0,0 +1,121 @@
+Mocha allows you to define and use custom reporters install via `npm`.
+
+For example, if `mocha-foo-reporter` was published to the npm registry, you could install it via `npm install mocha-foo-reporter --save-dev`, then use it via `mocha --reporter mocha-foo-reporter`.
+
+## Example Custom Reporter
+
+If you're looking to get started quickly, here's an example of a custom reporter:
+
+```js
+// my-reporter.js
+
+const Mocha = require('mocha');
+const {
+  EVENT_RUN_BEGIN,
+  EVENT_RUN_END,
+  EVENT_TEST_FAIL,
+  EVENT_TEST_PASS,
+  EVENT_SUITE_BEGIN,
+  EVENT_SUITE_END
+} = Mocha.Runner.constants;
+
+// this reporter outputs test results, indenting two spaces per suite
+class MyReporter {
+  constructor(runner) {
+    super(runner);
+
+    this._indents = 0;
+
+    runner
+      .once(EVENT_RUN_BEGIN, () => {
+        console.log('start');
+      })
+      .on(EVENT_SUITE_BEGIN, () => {
+        this.increaseIndent();
+      })
+      .on(EVENT_SUITE_END, () => {
+        this.decreaseIndent();
+      })
+      .on(EVENT_TEST_PASS, test => {
+        // Test#fullTitle() returns the suite name(s)
+        // prepended to the test title
+        console.log(`${this.indent()}pass: ${test.fullTitle()}`);
+      })
+      .on(EVENT_TEST_FAIL, (test, err) => {
+        console.log(
+          `${this.indent()}fail: ${test.fullTitle()} - error: ${err.message}`
+        );
+      })
+      .once(EVENT_RUN_END, () => {
+        console.log(
+          `end: ${this.stats.passes}/${this.stats.passes +
+            this.stats.failures} ok`
+        );
+      });
+  }
+
+  indent() {
+    return Array(this._indents)
+      .fill('  ')
+      .join('');
+  }
+
+  increaseIndent() {
+    this._indents++;
+  }
+
+  decreaseIndent() {
+    this._indents--;
+  }
+}
+
+module.exports = MyReporter;
+```
+
+To use this reporter, execute `mocha --reporter /path/to/my-reporter.js`.
+
+For further examples, the built-in reporter implementations are the [best place to look](https://github.com/mochajs/mocha/tree/master/lib/reporters). The [integration tests](https://github.com/mochajs/mocha/tree/master/test/reporters) may also be helpful.
+
+## The `Base` Reporter Class
+
+[Base] is an "abstract" class. It contains console-specific settings and utilities, commonly used by built-in reporters. Browsing the built-in reporter implementations, you may often see static properties of [Base] referenced.
+
+It's often useful--but not necessary--for a custom reporter to extend [Base].
```

Comment:
Add spaces around your dbl-hyphens so spelling can be checked more easily

Reply:
I've never heard of that issue before, can you explain more?

I can say with certainty that I don't love double-hyphens!  Maybe there's a smart hyphen transformation...

Reply:
Would prefer:
`It's often useful--but not necessary--for`
changed to:
`It's often useful -- but not necessary -- for`
or
`It's often useful (but not necessary) for`